modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/simple/GMLWriter.java

```diff
@@ -291,9 +333,14 @@ void coordinates(CoordinateSequence coordinates, char cs, char ts, StringBuffer
         for (int i = 0; i < n; i++) {
             coordFormatter.format(coordinates.getX(i), sb).append(cs);
             coordFormatter.format(coordinates.getY(i), sb);
-            if (dim == 3) {
-                sb.append(cs);
-                coordFormatter.format(coordinates.getOrdinate(i, 2), sb);
+            if (dim > 2) {
+                int totalDimensions =
```

Comment:
This idea was referenced as the "geometryDimension" in SFSQL, that is the dimension of just the spatial ordinates. So XYM data would have dimension 3 and geometry dimension of 2.